f5lbaasdriver/test/tempest/services/clients/bigip_client.py

```diff
@@ -196,3 +196,34 @@ def virtual_server_has_policy(self, vs_name, policy_name, partition):
                     return True
 
         return False
+
+    def virtual_server_has_profile(self, vs_name, profile_name, partition):
+        if self.virtual_server_exists(name=vs_name, partition=partition):
```

Comment:
We could just raise if no vs exists (or just let the sdk raise its exception when this happens) instead of making the if check. I guess if you called this method, the virtual server should exist at least. If it doesn't, we'd want to know why. We'd only get a False back from this call, and it might hide the fact that the vs didn't exist at all.